modules/augment/src/resize.cpp

```diff
@@ -0,0 +1,43 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+namespace cv { namespace augment {
+
+Resize::Resize(Size size,const std::vector<int>& interpolations)
+{
+    this->size = size;
+    if (interpolations.size() > 0)
+        this->interpolations = interpolations;
+    else
+        this->interpolations = { INTER_NEAREST , INTER_LINEAR , INTER_AREA , INTER_CUBIC , INTER_LANCZOS4 };
+}
+
+Resize::Resize(Size size, int interpolation)
+{
+    this->size = size;
+    interpolations = { interpolation };
+}
+
+void Resize::init(const Mat& srcImage)
+{
+    Transform::init(srcImage);
+    int index = Transform::rng.uniform(0, interpolations.size());
+    interpolation = interpolations[index];
+}
+
+void Resize::image(InputArray src, OutputArray dst)
+{
+    resize(src, dst, size, 0, 0, interpolation);
+}
+
+Point2f Resize::point(const Point2f& src)
+{
+    float x, y;
+    x = (src.x / srcImageCols)*size.width;
+    y = (src.y / srcImageRows)*size.height;
```

Comment:
If source input width is 4 and output is 8, for `x == 3` we'll have `x' = 8 * (3/4) = 6`. However 7 is expected.

Reply:
how is the expected 7? I tried it on multiple images and it maps the ground truth data to the correct place

Reply:
the point at 3 means that the landmark is at 3/4 from the image which maps to 6/8

Reply:
Oh, I understand it now.. sorry for the misunderstanding

Reply:
fixed, thanks.